core/modules/qproc/testQueryAnalysis.cc

```diff
@@ -814,6 +814,17 @@ BOOST_AUTO_TEST_CASE(dm646) {
     testAndCompare(qsTest, stmt, expected);
 }
 
+BOOST_AUTO_TEST_CASE(dm681) {
+    std::string stmt = "SELECT foo FROM Filter f limit 5";
+    std::string expected = "SELECT foo FROM LSST.Filter AS f LIMIT 5";
+    testAndCompare(qsTest, stmt, expected);
+
+    stmt = "SELECT foo from Filter f limit 5 garbage query !#$%!#$";
+    char const expectedErr[] = "ParseException:Parse token mismatch error:expecting EOF, found 'garbage':";
+    boost::shared_ptr<QuerySession> qs = testStmt3(qsTest, stmt, expectedErr);
+
+}
+
```

Comment:
Maybe add some regression checks for supported trailing semicolon scenarios, per comment above?